### DISCUSSION TITLE
IPixelCollection  change

### DISCUSSION THREAD

**Discussion by derekjohn37** [dlemstra/Magick.NET#790] [Help] [answered]
Following an update of Magick.NET, IPixelCollection now requires a struct. Are there predefined structures or do I have to define my own? Any advice will be appreciated.

Current code (not working)
using (IPixelCollection pixels = cmykImage.GetPixels())

Required code ?????? = struct
using (IPixelCollection<???????> pixels = cmykImage.GetPixels())

**Comment by dlemstra** [comment, marked as answer]
The `IPixelCollection` now requires this because of breaking changes in the API to make it possible to move some code to new NuGet packages. The struct that you should be using depends on the version of Magick.NET that you are using. It should be the same time as the one `MagickImage` is using in it's interface. I would advise you to use `var` instead to make your life a bit easier.

**Reply by derekjohn37** [reply]
Thanks dlemstra, using vars did the trick.
